### DISCUSSION TITLE
Mesh penetration problem in SOFA soft body simulation

### DISCUSSION THREAD

**Discussion by yangwxb** [InfinyTech3D/SofaUnity#122] [Help - Setting up SofaUnity]
Hello, I was simulating a hose device inserted into another mesh model and colliding at the edges of the model (hoses are rigid models), but the end result is that mesh penetration problems can easily occur. Do you have any good suggestions?

https://github.com/user-attachments/assets/b040e544-80d5-4f28-86ab-2b891b87f63c

**Comment by epernod** [comment]
Hi,

it would be useful to see the collision model you are using. Is it surface or volume mesh?
The collision model of your device might be too rough.

Best regards,

**Reply by yangwxb** [reply]
Hi，the collision model I use is the surface mesh. Do you mean that the model mesh loaded is too rough?

**Reply by epernod** [reply]
This might be that or your collision distance are too small compare to the motion you are doing. 
What collision pipeline are you using and collision models?
